### PR TITLE
Enable proxyauth and fix some small typos

### PR DESCRIPTION
**Pull request by jimklimov**
Proxyauth verified against a Sun Messaging server : 
````
 ./isbg.py --imapuser mailadmin --imappasswd Adm1nPaSs --imapproxyuser enduser@domain.com --imaplist --imaphost imap.domain.com

  "/" INBOX', '  "/" Drafts', '  "/" Junk', '  "/" Sent', '  "/" trusted'])

('select', 'INBOX.spam', 1) returned ('NO', ['Mailbox does not exist']) - aborting

Use --help to see valid options and arguments
````

Some more improvements piled on as I was beginning to actually use the script ;)

### CODE REVIEW

**Comment by baldurmen** [comment]
Hi!

Very nice of you to submit patches for new features.

Thing is, I've been thinking for a while now and I think we should aim for a complete rewrite of isbg.

The current code base is old and my previous attempts to make it more modern kinda failed.

So yeah, I won't be merging this now :s. Sorry.

But I won't close this either! When a new and shiny version of isbg will be written, I'll look at this.

**Comment by baldurmen** [comment]
And I think commit 43c64ce (shipping docopt) is a bad idea. Proper practice should be using either a version installed via pip, or preferably the packaged version for your OS.

Shipping docopt means we'd have to keep that code updated and frankly, I don't want to have to do that.

**Comment by jimklimov** [comment]
Regarding docopt - I'm not sure it must be shipped either, but it is not
preinstalled in e.g. Solaris (and the production servers I dealt with are
shining fresh Solaris 10 with newest of the line Python 2.4 ;) )... It's
not like the Solaris 8 boxes stopped working, just the script does not have
to run there ;) And there the Python is a monolith package, no hordes of
additional module-packages. You have what you are given.

So the options are that either the user has to google for a copy of docopt
and copy into /usr/lib/whatever or a home directory, or that they can take
chances with a provided copy that worked in at least a couple of cases ;)
It might be put into some contrib/ subdirectory to match other OSS project
layouts, but having them in the same directory allows to
un-tarball/git-clone the project and run instantly.

As for merging vs. not - it is your call, the code is there in my branch if
I were to package something for those boxes... but I'd rather package
upstream codes without local mods :) Besides, if these changes do not
explode, there's little harm in adding them while you plan out the rewrite
- some of my "TODOs" took literally decades to fulfill or abandon, and in
the meanwhile having something is better than nothing - IMHO.

Is there some pressing problem that would be solved by a full rewrite? So
far this script seems to work on even antique systems, and that is a good
thing for my use-case ;)

2016-11-28 6:25 GMT+01:00 Louis-Philippe Véronneau <notifications@github.com

**Comment by baldurmen** [comment]
I guess your right about merging. Could you rebase your commits without the doctopt addition so I can merge this during the Christmas holidays?

**Comment by jimklimov** [comment]
Ok, chopped this commit out (and added a reference into the README for people who might need it).
FWIW, addition of docopt is resubmitted as PR #62 so people might find it more easily ;)

File: README.md [hunk 1]
```diff
@@ -172,6 +172,13 @@ Options:
     --imappasswd passwd  IMAP account password
     --imapport port      Use a custom port
     --imapuser username  Who you login as
+    --imapproxyuser userbox     Use proxyauth on IMAP host. If set, allows
```
Comment (baldurmen) [inline README.md]: userbox should be changed to mbox to fit with the previously used syntax

File: isbg.py [hunk 2]
```diff
@@ -31,6 +31,13 @@
     --imappasswd passwd  IMAP account password
     --imapport port      Use a custom port
     --imapuser username  Who you login as
+    --imapproxyuser userbox     Use proxyauth on IMAP host. If set, allows
```
Comment (baldurmen) [inline isbg.py]: same comment as imapproxyuser

File: isbg.py [hunk 3]
```diff
@@ -299,7 +299,7 @@ def dehexof(x):
         imapport = 993
 
 if pastuidsfile is None:
-    pastuidsfile = os.path.expanduser("~" + os.sep + ".isbg-track")
+    pastuidsfile = os.path.expanduser("~" + os.sep + ".isbg-track" + "%" + imapuser + "%" + imapproxyuser)
```
Comment (baldurmen) [inline isbg.py]: I think I'd prefer to use "-" instead of "%"

**Comment by baldurmen** [comment]
Actually, to make it simpler, I merged it all, and reverted 1 commit.

I reverted `cbbc5e941adc8ffe8cf822891f99ad0a5ad35b26`  because:

* If you did not use `--imapproxyuser`, it crashed (can't concatenate `str` and `None`)
* we already set track files based on the IMAP host, username and port number.

I know this means you can't use the `--imapproxyuser` parameters if you want to run on multiple accounts in parallel, but I'll let you fix this :D 

IMHO, the way to go on this is to take `imapproxyuser` in account when creating the 16 byte identifier based on the IMAP host, username and port number.

File: isbg.py (hunk 3 above)
Comment (jimklimov) [inline isbg.py]: I thought about chars to use for separators - a percent is less likely in a user (or account) name than a dash, and the latter might lead to ambiguous conflicts, e.g. `.isbg-track-smith-wesson-admin` : which is the user and the admin part? what if your domain has both `smith`, `smith-wesson`, `wesson-admin` and `admin` accounts, and they happen to be active for isbg at the same time?

Possibly a contrived example, but when things like these run for a server with thousands of boxes, the law of big numbers kicks in and causes problems with non-negligible probability :) An improbable username character does away with this.

File: README.md (hunk 1 above)
Comment (jimklimov) [inline README.md]: I looked around, and the help text for "mbox" implies that this is a folder name (e.g. an "INBOX").

Here for proxyuser case, this is a user account name (e.g. the boss whose messages you manage while authenticated as a secretary's "username" with his "passwd", or you clean up a random email user while authenticated as a mail server admin).

Pending PR with "userbox" changed to "enduser" for clarity (hopefully).

**Pull request merged**